string config_path  # path to yaml file
---
# TODO solver type

float64[] initial_state

geometry_msgs/Vector3 gravity

# weights
upright_msgs/FloatArray input_weight
upright_msgs/FloatArray state_weight
upright_msgs/FloatArray end_effector_weight

# input limits
float64[] input_limit_lower
float64[] input_limit_upper
float64 input_limit_mu
float64 input_limit_delta

# state limits
float64[] state_limit_lower
float64[] state_limit_upper
float64 state_limit_mu
float64 state_limit_delta

# tracking gain
upright_msgs/FloatArray Kp

# operating trajectory
bool use_operating_points
# ocs2_msgs/mpc_target_trajectories operating_trajectory
float64[] operating_times
upright_msgs/FloatArray operating_states
upright_msgs/FloatArray operating_inputs

# URDF paths
string robot_urdf_path
string obstacle_urdf_path

# OCS2 settings
string ocs2_config_path
string lib_folder

# robot settings
string robot_base_type
string end_effector_link_name
upright_msgs/RobotDimensions dims

# TODO remaining settings
upright_msgs/TrayBalanceSettings tray_balance_settings
upright_msgs/InertialAlignmentSettings inertial_alignment_settings
